### ISSUE TITLE
Use generic driver, instead of virtual box to install docker swarm

### ISSUE THREAD

**Issue by patricktslee** [vfarcic/docker-flow#14]
Hi all,

I try to build a docker-machine with swarm using generic driver, it failed. Here is the error message.  Have you met it before?

`docker-machine create -d generic --generic-ip-address 10.0.0.246 --generic-ssh-key $HOME/.ssh/id_rsa --generic-ssh-user patrick --generic-ssh-port 22 --swarm --swarm-master --swarm-discovery="consul://$(docker-machine ip consul-machine):8500" --engine-opt="cluster-store=consul://$(docker-machine ip consul-machine):8500" --engine-opt="cluster-advertise=eth1:2376" swarm-master
`

Here is the error message.
**Running pre-create checks...
Creating machine...
(swarm-master) Importing SSH key...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Installing Docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Configuring swarm...
Error creating machine: Error running provisioning: Unable to pull image: Post https://10.0.0.246:2376/v1.15/images/create?fromImage=swarm%3Alatest: read tcp 10.0.0.113:52811->10.0.0.246:2376: read: connection reset by peer**

**Comment by lincze** [comment]
i gave up docker-machine, for the moment, too much frictions.

there exists a -D flag, it provides more info, what's going on:

docker-machine -D create -d...

**Comment by patricktslee** [comment]
Thanks. 

Any news tool you use to manage and install docker swarm?

**Comment by vfarcic** [comment]
I never used Docker Machine with the generic driver. To be more precise, I use it only for demo purposes (simulating cluster locally). In production, I tend to use Ansible for the initial (infrastructure) setup when running on-premise and Packer/Ansible/Terraform when using cloud providers (i.e. AWS). While Docker Machine is useful, in my case, it is not sufficient for production use.

I'm sure that my answer/comments do not do much good. We can take a look at the problem together through Skype or HangOuts.

**Comment by patricktslee** [comment]
Eventually I fixed it.  I noticed that the interface name is incorrect on my command.  Because I used Ubuntu on my virtual box.  The Interface name is enp0s3 not eth1.  So I change the command as below.

docker-machine -D create -d generic --generic-ip-address 10.0.0.246 --generic-ssh-key $HOME/.ssh/id_rsa --generic-ssh-user patrick --generic-ssh-port 22 --swarm --swarm-master --swarm-discovery="consul://$(docker-machine ip consul-machine):8500" --engine-opt="cluster-store=consul://$(docker-machine ip consul-machine):8500" --engine-opt="cluster-advertise=**enp0s3**:2376" swarm-master

And when I list docker machine.  It show the swarm-master.
docker-machine ls
NAME             ACTIVE   DRIVER    STATE     URL                     SWARM                   DOCKER    ERRORS
consul-machine   *        generic   Running   tcp://10.0.0.230:2376                           v1.11.2   
machine1         -        generic   Running   tcp://10.0.0.240:2376                           v1.11.2   
swarm-master     -        generic   Running   tcp://10.0.0.246:2376   swarm-master (master)   v1.11.2   

Thanks your guideline and the -D flag is very useful.
